### PR TITLE
New and improved docs for AzureMaintenanceEvent pub/sub channel

### PR DESCRIPTION
**Pull request by amsoedal**
- Added new maintenance events
- Added more context/background info
- Got rid of out of date info

### CODE REVIEW

File: AzureRedisEvents.md [hunk 1]
```diff
@@ -1,143 +1,81 @@
...
+## Subscribing to the channel
+
+If you are using C#, we recommend relying on the StackExchange.Redis library. It will automatically subscribe to the pub/sub channel and raise the ServerMaintenanceEvent in response to maintenance. For more information, please see the [StackExchange.Redis documentation](https://github.com/StackExchange/StackExchange.Redis/blob/main/docs/ServerMaintenanceEvent.md).
+
+For other client libraries, you will need to subscribe to the AzureRedisEvents pub/sub channel and implement logic to parse incoming maintenance event messages.
 
 ## Message format
 
-Each entry in the message is separated by using a pipe (|) as a delimiter. All messages start with a *FieldName* followed by the *Entry* for the field and so on so forth.
+Messages sent by Azure Redis are pipe (|) delimited strings. All messages start with a *FieldName* followed by the *Entry* for the field and so on so forth.
```
Comment (philon-msft) [inline AzureRedisEvents.md:38]: "All messages start with a *FieldName* followed by the *Entry* for the field, followed by additional pairs of field names and entries"

File: AzureRedisEvents.md [hunk 2]
```diff
@@ -1,143 +1,81 @@
...
-Similarly, there will be a notification message that is received when the maintenance ends that will sent through the *AzureRedisEvents* channel. You do *NOT* need to wait for this message to use the LB endpoint. The LB endpoint is always available. However, we included this for logging purposes or for customers who use the replica endpoint in clusters for read workloads.
-
 ## Walking through a sample maintenance event
 
-1. App is connected to Redis and everything is working fine. 
-
-2. Current Time: [16:25:40] -> Message received through *AzureRedisEvents* channel. The message notification type is "NodeMaintenanceStarting" and StartTimeInUTC is "16:26:10" (about 30 seconds from current time). So we wait. 
-
-        NotificationType|NodeMaintenanceStarting|StartTimeInUTC|2020-10-14T16:26:10|IsReplica|False|IPAddress|52.158.249.185|SSLPort|15001|NonSSLPort|13001
+1. App is connected to Redis and everything is working fine.
```
Comment (philon-msft) [inline AzureRedisEvents.md:46]: "and everything is functioning normally"

File: AzureRedisEvents.md [hunk 3]
```diff
@@ -1,143 +1,81 @@
...
+    * Note: the start time for this event is an approximation, because we will start getting ready for the update proactively and the node may become unavailable up to 3 minutes sooner. We recommend listening for `NodeMaintenanceStarting` and `NodeMaintenanceStart` for the highest level of accuracy (these are only likely to differ by a few seconds at most).
+3. Current Time: [16:34:26] -> `NodeMaintenanceStarting` message is received, and `StartTimeInUTC` is 16:34:46, about 20 seconds from the current time.
+4. Current Time: [16:34:46] -> `NodeMaintenanceStart` message is received, so we know the node maintenance is about to happen. We break the circuit and stop sending new operations to the Redis connection. (Note: the appropriate action for your application may be different.)
+5. Current Time: [16:34:47] -> The connection is closed by the Redis server.
+6. Current Time: [16:34:56] -> `NodeMaintenanceFailoverComplete` message is received. This tells us that the replica node has promoted itself to primary, so the other node can go offline for maintenance.
```
Comment (philon-msft) [inline AzureRedisEvents.md:52]: Mention that not all maintenance events will cause a failover

File: AzureRedisEvents.md [hunk 4]
```diff
@@ -1,143 +1,81 @@
...
 
-4. Current Time: [16:26:10] -> The Redis object is disconnected from the Redis server. You can listen to these ConnectionDisconnected events from most clients [StackExchange Events](<https://stackexchange.github.io/StackExchange.Redis/Events>) or [Lettuce Events](<https://github.com/lettuce-io/lettuce-core/wiki/Connection-Events#connection-events>).
+##  Event details
 
-5. Current Time [16:26:10] -> The Redis object is reconnected back to the Redis server (again, you can listen to the Reconnected event on your client). It is safe to send ops again to the Redis connection and all ops will succeed.
+#### NodeMaintenanceScheduled event
 
-6. Current Time [16:27:42] -> Message received through *AzureRedisEvents* channel. The message notification type is "NodeMaintenanceEnded" and StartTimeInUTC is "16:27:42". Nothing to do here if you are talking to 6380/6379. For clustered caches, you can start sending readonly workloads to the replica. 
+`NodeMaintenanceScheduled` events are raised for maintenance scheduled by Azure, up to 15 minutes in advance. This event will not get fired for user-initiated reboots.
```
Comment (philon-msft) [inline AzureRedisEvents.md:61]: "will not be fired"

Comment (philon-msft) [inline AzureRedisEvents.md:61]: Also note that "scheduled" won't be fired for monthly patching, just infrastructure maintenance?

**Review by philon-msft — APPROVED** [review]

File: AzureRedisEvents.md (hunk 3 above)
Comment (amsoedal) [reply AzureRedisEvents.md:52]: Good catch

File: AzureRedisEvents.md (hunk 4 above)
Comment (amsoedal) [reply AzureRedisEvents.md:61]: Yeah good call

**Pull request merged**